sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/descriptor/MetricDescriptor.java

```diff
@@ -41,12 +55,26 @@ public static MetricDescriptor create(View view, InstrumentDescriptor instrument
 
   public abstract String getUnit();
 
+  /** The view that lead to the creation of this metric, if applicable. */
+  public abstract Optional<View> getSourceView();
+  /** The instrument which lead to the creation of this metric. */
+  public abstract InstrumentDescriptor getSourceInstrument();
+
   @Memoized
   @Override
   public abstract int hashCode();
 
+  /** Returns true if another metric descriptor is compatible with this one. */
   public boolean isCompatibleWith(MetricDescriptor other) {
-    // TODO: implement.
-    return equals(other);
+    return sameString(getName(), other.getName())
+        && sameString(getDescription(), other.getDescription())
+        && sameString(getUnit(), other.getUnit());
+  }
+
+  private static boolean sameString(String lhs, String rhs) {
```

Comment:
`Objects.equals(lhs, rhs)` ?

Reply:
Same, my java is pretty rusty, thanks for the call out!